### PR TITLE
Add config for auto place tiles

### PR DESCRIPTION
**Pull request by sram1337**
Adds a configuration for toggling auto-marking tiles.

I also refactored markTile. I split out the logic for handling marking during movement vs from the option menu into two functions: `handleMenuOption` and `handleMovement`. I also created a function `updateTileMark` which just handles marking/unmarking a single tile. `updateTileMark` takes an explicit mark/unmark boolean param and will try to add/remove the tile based on that. I did this to try to fix some weirdness that occured when both the groundMarker and tilemanMode plugins are on, but it didn't fully fix the issue.

### CODE REVIEW

File: src/main/java/com/tileman/TilemanModePlugin.java [hunk 1]
```diff
@@ -247,42 +247,57 @@ int getRemainingTiles() {
         return remainingTiles;
     }
 
-    void markTile(LocalPoint localPoint, boolean menuOption) {
-        if (localPoint == null || client.isInInstancedRegion() || (lastTile != null && lastTile.distanceTo(localPoint) == 0)) {
```
Comment (ConorLeckey) [inline src/main/java/com/tileman/TilemanModePlugin.java:251]: Should we still not check the tile is different with
`(lastTile != null && lastTile.distanceTo(localPoint) == 0)`

File: src/main/java/com/tileman/TilemanModeConfig.java [hunk 2]
```diff
@@ -78,4 +78,16 @@ default int tilesOffset()
...
+	@Alpha
+	@ConfigItem(
+			keyName = "automarkTiles",
+			name = "Auto-mark tiles",
+			description = "Automatically mark tiles as you walk.",
+			warning = "Will sometimes miss marking a tile"
+	)
+	default boolean automarkTiles()
+	{
+		return false;
```
Comment (ConorLeckey) [inline src/main/java/com/tileman/TilemanModeConfig.java:91]: Should this default to true?

**Comment by ConorLeckey** [comment]
Looks good to merge

File: src/main/java/com/tileman/TilemanModeConfig.java (hunk 2 above)
Comment (sram1337) [reply src/main/java/com/tileman/TilemanModeConfig.java:91]: auto mark defaults to false so new accounts aren't spending their tiles on misclicks

**Pull request merged**